File: Kotlin/프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.kt
class Solution {
	fun solution(n: Int, m: Int): IntArray {
		var answer = IntArray(2)

		val gcd = GCD(n, m)
		
		answer[0] = gcd
		answer[1] = n * m / gcd;
		
		return answer
	}

	fun GCD(a: Int, b: Int): Int {
		return if (a % b == 0) {
			b
		} else {
			GCD(b, a%b)
		}
	}
}

File: JavaScript/프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    var answer = [];

    const gcd = getGCD(n, m);

    const lcm = n * m / gcd;

    return [gcd, lcm];
}

function getGCD(n, m) {
    if (n % m === 0) {
        return m;
    } else {
        console.log(m, n);
        return getGCD(m, n % m)
    }
}